externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/schema/RowSet.scala

```diff
@@ -0,0 +1,371 @@
+      } else {
+        ret.add(idx, row(ordinal).asInstanceOf[T])
+      }
+      idx += 1
+    }
+    ret
+  }
+
+  import scala.language.implicitConversions
+
+  implicit private def bitSetToBuffer(bitSet: java.util.BitSet): ByteBuffer = {
+    ByteBuffer.wrap(bitSet.toByteArray)
+  }
+
+  private def toTColumnValue(
+      ordinal: Int,
+      row: List[Any],
+      types: List[Column],
+      timeZone: ZoneId): TColumnValue = {
+
+    types(ordinal).getType match {
+      case BOOLEAN =>
+        val boolValue = new TBoolValue
+        if (row(ordinal) != null) boolValue.setValue(row(ordinal).asInstanceOf[Boolean])
+        TColumnValue.boolVal(boolValue)
+
+      case TINYINT =>
+        val byteValue = new TByteValue
+        if (row(ordinal) != null) byteValue.setValue(row(ordinal).asInstanceOf[Byte])
+        TColumnValue.byteVal(byteValue)
+
+      case SMALLINT =>
+        val tI16Value = new TI16Value
+        if (row(ordinal) != null) tI16Value.setValue(row(ordinal).asInstanceOf[Short])
+        TColumnValue.i16Val(tI16Value)
+
+      case INTEGER =>
+        val tI32Value = new TI32Value
+        if (row(ordinal) != null) tI32Value.setValue(row(ordinal).asInstanceOf[Int])
+        TColumnValue.i32Val(tI32Value)
+
+      case BIGINT =>
+        val tI64Value = new TI64Value
+        if (row(ordinal) != null) tI64Value.setValue(row(ordinal).asInstanceOf[Long])
+        TColumnValue.i64Val(tI64Value)
+
+      case REAL =>
+        val tDoubleValue = new TDoubleValue
+        if (row(ordinal) != null) tDoubleValue.setValue(row(ordinal).asInstanceOf[Float])
+        TColumnValue.doubleVal(tDoubleValue)
+
+      case DOUBLE =>
+        val tDoubleValue = new TDoubleValue
+        if (row(ordinal) != null) tDoubleValue.setValue(row(ordinal).asInstanceOf[Double])
+        TColumnValue.doubleVal(tDoubleValue)
+
+      case VARCHAR =>
+        val tStringValue = new TStringValue
+        if (row(ordinal) != null) tStringValue.setValue(row(ordinal).asInstanceOf[String])
+        TColumnValue.stringVal(tStringValue)
+
+      case _ =>
+        val tStrValue = new TStringValue
+        if (row(ordinal) != null) {
+          tStrValue.setValue(
+            toHiveString((row(ordinal), types(ordinal).getType), timeZone))
+        }
+        TColumnValue.stringVal(tStrValue)
+    }
+  }
+
+  private def createBuilder(): DateTimeFormatterBuilder = {
+    new DateTimeFormatterBuilder().parseCaseInsensitive()
+  }
+
+  private lazy val dateFormatter = {
+    createBuilder().appendPattern("yyyy-MM-dd")
+      .toFormatter(Locale.US)
+      .withChronology(IsoChronology.INSTANCE)
+  }
+
+  private lazy val simpleDateFormatter = new SimpleDateFormat("yyyy-MM-dd", Locale.US)
+
+  private lazy val timeFormatter: DateTimeFormatter = {
+    createBuilder().appendPattern("HH:mm:ss")
+      .toFormatter(Locale.US)
+      .withChronology(IsoChronology.INSTANCE)
+  }
+
+  private lazy val simpleTimeFormatter = new SimpleDateFormat("HH:mm:ss", Locale.US)
```

Comment:
We should avoid using `SimpleDateFormat`, please use `java.time` classes if possible, otherwise, use `FastDateFormat` provided by `commons-lang3`, which is thread-safe and fast than `SimpleDateFormat`